### PR TITLE
multi: update dex client

### LINKED ISSUE #632: dcrdex: Update client version.
image

### PR DESCRIPTION
**Pull request by ukane-philemon**
Closes #632

### CODE REVIEW

**Comment by ukane-philemon** [comment]
I've noticed a few issues that I'm unable to resolve atm:

1. The server and market selection on the dex market page no longer stack on top of every element on the page when the dropdown is expanded
2. The dex market page doesn't scroll downwards unless the scrollbar is used (noticed the same thing on the post bond step).
3. The order type selection on the order form should be directly under the order type collapsed view.

I think issues arose after recent dropdown code changes and the gioui version update.

**Comment by bugjt** [comment]
I have solved it

> I've noticed a few issues that I'm unable to resolve atm:
> 
> 1. The server and market selection on the dex market page no longer stack on top of every element on the page when the dropdown is expanded
> 2. The dex market page doesn't scroll downwards unless the scrollbar is used (noticed the same thing on the post bond step).
> 3. The order type selection on the order form should be directly under the order type collapsed view.
> 
> I think these issues arose after recent dropdown code changes and the gioui version update.

File: ui/cryptomaterial/dropdown.go [hunk 1]
```diff
@@ -316,6 +316,7 @@ func (d *DropDown) collapsedAndExpandedLayout(gtx C) D {
 // expandedLayout computes dropdown layout when dropdown is opened.
 func (d *DropDown) expandedLayout(gtx C) D {
 	m := op.Record(gtx.Ops)
+	gtx.Constraints.Max.Y = inf
```
Comment (dreacot) [inline ui/cryptomaterial/dropdown.go:319]: what is inf? @bugjt

Comment (bugjt) [reply ui/cryptomaterial/dropdown.go:319]: This makes the dropdown menu display on the window size, regardless of the size of its parent that contains it.

Comment (bugjt) [reply ui/cryptomaterial/dropdown.go:319]: It fixes this bug: "1. The server and market selection on the dex market page no longer stack on top of every element on the page when the dropdown is expanded"

**Comment by ukane-philemon** [comment]
https://github.com/crypto-power/cryptopower/pull/633#issuecomment-2334526408 has been resolved by https://github.com/crypto-power/cryptopower/pull/633/commits/756a6cf52e188a2fb754f1cac9c1cf67ca4419a9

**Review by JoeGruffins — APPROVED** [review]
Trading is not working on dextest, I'll look into why not.

Edit, I found the logs, and trading is working I forgot to build with an adjusted locktime

**Comment by JoeGruffins** [comment]
I found the logs, its checking bonds every millisecond?

**Comment by JoeGruffins** [comment]
Seeing some weird stuff with bonds. It's spinning sowhere:
```
2024-09-19 08:31:58.934 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
2024-09-19 08:31:58.934 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
2024-09-19 08:31:58.945 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
2024-09-19 08:31:58.946 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
2024-09-19 08:31:58.946 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
2024-09-19 08:31:58.946 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
2024-09-19 08:31:58.946 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
2024-09-19 08:31:58.957 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
2024-09-19 08:31:58.958 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
2024-09-19 08:31:58.958 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
2024-09-19 08:31:58.958 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
2024-09-19 08:31:58.958 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
2024-09-19 08:31:58.973 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
2024-09-19 08:31:58.974 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
2024-09-19 08:31:58.974 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
```

**Comment by ukane-philemon** [comment]
> Seeing some weird stuff with bonds. It's spinning somewhere:
> 
> ```
> 2024-09-19 08:31:58.934 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
> 2024-09-19 08:31:58.934 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
> 2024-09-19 08:31:58.945 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
> 2024-09-19 08:31:58.946 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
> 2024-09-19 08:31:58.946 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
> 2024-09-19 08:31:58.946 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
> 2024-09-19 08:31:58.946 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
> 2024-09-19 08:31:58.957 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
> 2024-09-19 08:31:58.958 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
> 2024-09-19 08:31:58.958 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
> 2024-09-19 08:31:58.958 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
> 2024-09-19 08:31:58.958 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
> 2024-09-19 08:31:58.973 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
> 2024-09-19 08:31:58.974 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
> 2024-09-19 08:31:58.974 [DBG] DEXC: Soon to expire bond found: acec8ef3f7b76c10742240bbbd02b5120e813e0b2a60aa4ee4a08d38f2c329d9:0 (dcr)
> ```

I've investigated this. 

dcrdex checks for bond rotation every `bondTickInterval` (`20 secs`) in [watchBonds](https://github.com/decred/dcrdex/blob/ae2fbd0eb2dd56eadf5095c21c663957fb083a26/client/core/bond.go#L47-#L61)  and continues to log this until an okay bond has been posted. See: [watchBonds](https://github.com/decred/dcrdex/blob/ae2fbd0eb2dd56eadf5095c21c663957fb083a26/client/core/bond.go#L47-#L61)  -> [bondStateOfDex](https://github.com/decred/dcrdex/blob/ae2fbd0eb2dd56eadf5095c21c663957fb083a26/client/core/bond.go#L714) -> [where the log actually happens](https://github.com/decred/dcrdex/blob/ae2fbd0eb2dd56eadf5095c21c663957fb083a26/client/core/bond.go#L391-#L401)

Same thing happens when a newly expired bond is found.

**Comment by JoeGruffins** [comment]
> dcrdex checks for bond rotation every bondTickInterval (20 secs)

In my logs its logging 4 times per millisecond. It should only be once ever 20 seconds correct?

**Comment by JoeGruffins** [comment]
More odd logging, bot the number of times in a millisecond it happens and why it needs usdc.eth:
```
2024-09-24 06:16:15.007 [ERR] DEXC: couldn't find minimum lot size for usdc.eth
2024-09-24 06:16:15.007 [ERR] DEXC: couldn't find minimum lot size for usdc.eth
2024-09-24 06:16:15.007 [ERR] DEXC: couldn't find minimum lot size for usdt.eth
2024-09-24 06:16:15.007 [ERR] DEXC: couldn't find minimum lot size for usdt.eth
2024-09-24 06:16:15.008 [ERR] DEXC: couldn't find minimum lot size for eth
2024-09-24 06:16:15.008 [ERR] DEXC: couldn't find minimum lot size for usdc.eth
2024-09-24 06:16:15.008 [ERR] DEXC: couldn't find minimum lot size for usdc.eth
2024-09-24 06:16:15.008 [ERR] DEXC: couldn't find minimum lot size for usdt.eth
2024-09-24 06:16:15.008 [ERR] DEXC: couldn't find minimum lot size for usdt.eth
2024-09-24 06:16:15.008 [ERR] DEXC: couldn't find minimum lot size for eth
2024-09-24 06:16:15.008 [ERR] DEXC: couldn't find minimum lot size for usdc.eth
2024-09-24 06:16:15.008 [ERR] DEXC: couldn't find minimum lot size for usdc.eth
2024-09-24 06:16:15.008 [ERR] DEXC: couldn't find minimum lot size for usdt.eth
2024-09-24 06:16:15.008 [ERR] DEXC: couldn't find minimum lot size for usdt.eth
```

**Comment by ukane-philemon** [comment]
>. it looks like func (pg *DEXMarketPage) exchange() *core.Exchange is being hit many times per millisecond

Aha, thanks. that's the other place bondStateOfDex is being called causing the spamming logs.

**Comment by JoeGruffins** [comment]
It looks like something else is still being hammered:
```
2024-09-25 04:44:16.196 [TRC] DEXC: wallet status requested for unknown asset 42 -> dcr
2024-09-25 04:44:16.340 [TRC] DEXC: wallet status requested for unknown asset 42 -> dcr
2024-09-25 04:44:16.345 [TRC] DEXC: wallet status requested for unknown asset 42 -> dcr
2024-09-25 04:44:16.359 [TRC] DEXC: wallet status requested for unknown asset 42 -> dcr
2024-09-25 04:44:16.373 [TRC] DEXC: wallet status requested for unknown asset 42 -> dcr
2024-09-25 04:44:16.401 [TRC] DEXC: wallet status requested for unknown asset 42 -> dcr
2024-09-25 04:44:16.425 [TRC] DEXC: wallet status requested for unknown asset 42 -> dcr
2024-09-25 04:44:16.453 [TRC] DEXC: wallet status requested for unknown asset 42 -> dcr
2024-09-25 04:44:16.453 [TRC] DEXC: wallet status requested for unknown asset 42 -> dcr
2024-09-25 04:44:16.479 [TRC] DEXC: wallet status requested for unknown asset 42 -> dcr
2024-09-25 04:44:16.863 [TRC] DEXC: wallet status requested for unknown asset 42 -> dcr
2024-09-25 04:44:16.870 [TRC] DEXC: wallet status requested for unknown asset 42 -> dcr
2024-09-25 04:44:16.907 [TRC] DEXC: wallet status requested for unknown asset 42 -> dcr
```

**Comment by JoeGruffins** [comment]
When I navigate to the dex onboarding page, it calls `func (pg *DEXOnboarding) bondAmountInfoDisplay(gtx C) D` and `func (pg *DEXOnboarding) bondAccountHasEnough() bool` about 30 times instantly. Do you know why? Is that just how it is? Seems like resources are being wasted. I guess because The logic is inside the view constructors? Is this correct?

**Review by JoeGruffins — APPROVED** [review]
I think there's lots of room for optimizing code but otherwise I think it looks good. Can optimize now or later.

**Review by dreacot — APPROVED** [review]

**Pull request merged**